### ISSUE TITLE
Checkbox

### ISSUE THREAD

**Issue by airikej** [TEHIK-EE/tedi-design-system#236]
https://github.com/TEHIK-EE/tedi-design-system/issues/122

[Figma](https://www.figma.com/design/jWiRIXhHRxwVdMSimKX2FF/%E2%9D%97TEDI-READY%E2%9D%97(follow-the-badges-in-header-to-see-which-components-are-ready-(designed%2Fdeveloped))?node-id=4228-72934&m=dev)

Dependencies:
- Grid
- Icon
- Tooltip

Changes:
- No API changes
- Add unit test
- Check WCAG

**Comment by zalizko** [comment]
Hello. I'll copy here my bug report from Slack channel (https://tehikustiilijuhis.slack.com/archives/C045VLXAXD3/p1733953510715499):

I found that currently <Check> component from comminity version of TEDI have a bug.
When form was hidden and enabled again - form reset don't work as expected.

Sandbox: https://codesandbox.io/p/sandbox/react-fiddle-forked-9ztst4.
Steps to reproduce:
1. click checkbox
2. click ‘Toggle’ twice (close and open form again)
3. click ‘Reset’

Nothing happens.

Also the following error occurred in the console: `Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?`
